pkg/cmd/server/api/validation/master.go

```diff
@@ -61,6 +61,13 @@ func ValidateMasterConfig(config *api.MasterConfig) ValidationResults {
 		validationResults.AddErrors(urlErrs...)
 	}
 
+	switch {
+	case config.ControllerLeaseTTL > 300:
+		validationResults.AddErrors(fielderrors.NewFieldInvalid("controllerLeaseTTL", config.ControllerLeaseTTL, "TTL must be -1 (disabled), 0 (default), or between 10 and 300 seconds"))
+	case config.ControllerLeaseTTL > 0 && config.ControllerLeaseTTL < 10:
+		validationResults.AddErrors(fielderrors.NewFieldInvalid("controllerLeaseTTL", config.ControllerLeaseTTL, "TTL must be -1 (disabled), 0 (default), or between 10 and 300 seconds"))
+	}
```

Comment:
You're right, sorry.

On Fri, Aug 14, 2015 at 4:11 PM, David Eads <notifications@github.com>
wrote:

> In pkg/cmd/server/api/validation/master.go
> <https://github.com/openshift/origin/pull/4137#discussion_r37114630>:
>
> > @@ -61,6 +61,13 @@ func ValidateMasterConfig(config *api.MasterConfig) ValidationResults {
> >  		validationResults.AddErrors(urlErrs...)
> >  	}
> >
> > +	switch {
> > +	case config.ControllerLeaseTTL > 300:
> > +		validationResults.AddErrors(fielderrors.NewFieldInvalid("controllerLeaseTTL", config.ControllerLeaseTTL, "TTL must be -1 (disabled), 0 (default), or between 10 and 300 seconds"))
> > +	case config.ControllerLeaseTTL > 0 && config.ControllerLeaseTTL < 10:
> > +		validationResults.AddErrors(fielderrors.NewFieldInvalid("controllerLeaseTTL", config.ControllerLeaseTTL, "TTL must be -1 (disabled), 0 (default), or between 10 and 300 seconds"))
> > +	}
>
> I've done a lot of reviews today, but I'm pretty sure this code allows
> -100 in conflict with your messages
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4137/files#r37114630>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift